### PR TITLE
Migrate timeline block

### PR DESCRIPTION
**Pull request by manishsongirkar**
Migrate the timeline block using Middleware js.

### CODE REVIEW

**Comment by sayedtaqui** [comment]
@manishsongirkar I see a little bit of broken UI

image

File: blocks/timeline/index.js [hunk 1]
```diff
@@ -26,139 +19,104 @@ registerBlockType( 'rtgb/timeline', {
...
 	attributes: {
 
-		timelineTitle: {
+		title: {
 			type: 'array',
 			source: 'children',
 			selector: '.timelineTitle',
+			field: {
+				type: 'rich-text',
+				className: 'timelineTitle',
```
Comment (sayedtaqui) [inline blocks/timeline/index.js]: Class name should be `timeline-title`.

File: blocks/timeline/index.js [hunk 2]
```diff
@@ -171,33 +129,16 @@ registerBlockType( 'rtgb/timeline', {
...
-								onChange={ onContentChange }
-								focus={ setFocus }
-								onFocus={ onFocusContent }
-							/>
+							{ middleware.fields.title }
+							{ middleware.fields.content }
 
 							<div className="postlinks">
-								{ ( newslink || bloglink ) ? <strong>Links: </strong> : null }
+								{ ( newsLink || blogLink ) ? <strong>Links: </strong> : null }
```
Comment (sayedtaqui) [inline blocks/timeline/index.js]: "Links" should be in translation function `__()`

File: blocks/timeline/index.js [hunk 3]
```diff
@@ -234,28 +175,26 @@ registerBlockType( 'rtgb/timeline', {
...
+							</h4>
+							<p className="timeline-description">
 								<span className="contributor">
-									{ timelineContent }
+									{ content }
 								</span>
 							</p>
 							<div className="postlinks">
-								{ ( newslink || bloglink ) ? <strong>Links: </strong> : null }
+								{ ( newsLink || blogLink ) ? <strong>Links: </strong> : null }
```
Comment (sayedtaqui) [inline blocks/timeline/index.js]: "Links" should be in `__()`

File: blocks/timeline/index.js [hunk 4]
```diff
@@ -234,28 +175,26 @@ registerBlockType( 'rtgb/timeline', {
...
 
 								{
-									newslink ? <a href={ newslink } className="timeline-news-link">{ __( 'News' ) }</a> : ''
+									newsLink ? <a href={ newsLink } className="timeline-news-link">{ __( 'News' ) }</a> : ''
```
Comment (sayedtaqui) [inline blocks/timeline/index.js]: Can also write it like this
```
!! newsLink && <a href={ newsLink } className="timeline-news-link">{ __( 'News' ) }</a>
```
Other conditions can also be modified like this.

File: blocks/timeline/index.js (hunk 1 above)
Comment (manishsongirkar) [reply blocks/timeline/index.js]: Done

File: blocks/timeline/index.js (hunk 2 above)
Comment (manishsongirkar) [reply blocks/timeline/index.js]: Done

File: blocks/timeline/index.js (hunk 3 above)
Comment (manishsongirkar) [reply blocks/timeline/index.js]: Done

File: blocks/timeline/index.js (hunk 4 above)
Comment (manishsongirkar) [reply blocks/timeline/index.js]: Done

**Pull request merged**